private/buf/cmd/buf/command/format/format.go

```diff
@@ -346,17 +347,25 @@ func run(
 				return err
 			}
 		}
-		// If we haven't overridden the output flag and havent set write, we can stop here.
+		// If we haven't overridden the output flag and haven't set write, we can stop here.
 		if flags.Output == "-" && !flags.Write {
 			return nil
 		}
 	}
 	if flags.Write {
+		changedPathSet := make(map[string]struct{}, len(changedPaths))
```

Comment:
nit: you can use `slicesext.ToStructMap(changedPaths)`

Reply:
done